### PR TITLE
more handling for core0 error to proppagate to api

### PR DESCRIPTION
**Pull request by abdulrahmantkhalifa**
this relates to issue https://github.com/zero-os/0-orchestrator/issues/1006
this depends on PR https://github.com/zero-os/0-core/pull/501

### CODE REVIEW

File: api/tools/ays.go [hunk 1]
```diff
@@ -119,24 +119,33 @@ func (aystool AYStool) WaitJobDone(jobid, repoName string) (ays.Job, error) {
...
+
+	err.err = fmt.Errorf(err.Message)
+	errResp := http.Response{
+		StatusCode: err.Code,
+	}
+	httperror := HTTPError{
+		Resp: &errResp,
+		err:  err.err,
+	}
+	fmt.Println(httperror)
```
Comment (khaledkbadr) [inline api/tools/ays.go]: remove or use log if logging was the purpose

File: api/tools/ays.go [hunk 2]
```diff
@@ -227,16 +236,16 @@ func (aystool AYStool) archiveBlueprint(blueprintName string, repoName string) e
 func (aystool AYStool) getRun(runid, repoName string) (*ays.AYSRun, error) {
 	run, resp, err := aystool.Ays.GetRun(runid, repoName, nil, nil)
 	if err != nil {
-		return nil, NewHTTPError(resp, err.Error())
+		return &run, NewHTTPError(resp, err.Error())
```
Comment (khaledkbadr) [inline api/tools/ays.go]: if error occurred, the run is useless here no?

File: api/tools/runs.go [hunk 3]
```diff
@@ -30,23 +32,48 @@ func WaitOnRun(api API, w http.ResponseWriter, r *http.Request, runid string) (R
...
+					break
+				}
+			}
 		}
-		return Run{}, err
 	}
+	if jobErr != nil {
+		httpErr, ok := jobErr.(HTTPError)
+		if ok {
+			fmt.Println(httpErr.Resp.StatusCode)
```
Comment (khaledkbadr) [inline api/tools/runs.go]: use log if logging was the purpose

File: api/tools/runs.go [hunk 4]
```diff
@@ -30,23 +32,48 @@ func WaitOnRun(api API, w http.ResponseWriter, r *http.Request, runid string) (R
...
+		errmsg := fmt.Sprintf("error waiting on job %s", job.Key)
+		WriteError(w, http.StatusInternalServerError, err, errmsg)
+		return Run{Runid: run.Key, State: EnumRunState(run.State)}, jobErr
+	}
+
 	if EnumRunState(runstatus.State) != EnumRunStateok {
 		err = fmt.Errorf("Internal Server Error")
 		WriteError(w, http.StatusInternalServerError, err, "")
-		return Run{}, err
+		return Run{Runid: run.Key, State: EnumRunState(run.State)}, jobErr
```
Comment (khaledkbadr) [inline api/tools/runs.go]: not used no?

File: api/tools/ays.go (hunk 2 above)
Comment (abdulrahmantkhalifa) [reply api/tools/ays.go]: so i can get the errored jobs

File: api/tools/runs.go (hunk 4 above)
Comment (abdulrahmantkhalifa) [reply api/tools/runs.go]: used cannot pass nil, and i might as well pass the actual run struct instead of an empty one

**Pull request merged**